### ISSUE TITLE
[CWM] About box support multiple language

### ISSUE THREAD

**Issue by lin-d-hop** [DigitalCommons/mykomap-monolith#85]

**Comment by lin-d-hop** [comment]
@ColmDC are there translations for this? Shall I make an actual issue?

**Comment by ColmDC** [comment]
>@ColmDC are there translations for this? Shall I make an actual issue?
Sure. I was going to flesh it out later.

Text needs to go here I believe. https://github.com/DigitalCommons/cwm-test-data/blob/delhi-demo/datasets/delhi/about.md

**Comment by ColmDC** [comment]
> are there translations for this? 

I have just been chasing that now...

**Comment by rogup** [comment]
It looks like the text currently being used is just plain and doesn't use any markdown syntax, which the app supports.

If we don't care about markdown (e.g. for section headings, bold, italics, etc.), a very easy way to do this would be to just add the about text in each language as a ui vocab string.

If we want to keep the option of using markdown syntax, we might need to make a folder of markdown files, one for each language, which can be served by the API.

Any preference for this? @ColmDC

**Comment by ColmDC** [comment]
I think having a seperate markdown file for each language seems to breaks with the multi-lang support in the rest of the application. Let's leave out the complication of MD support for now.

**Comment by ColmDC** [comment]
About box text for next CWM update.

-----
Welcome to the Cooperative World Map! 

The Cooperative World Map (CWM) is an interactive and up-to-date digital map that aspires to unite and showcase all cooperative entities in the world. The CWM is a visual representation of all cooperative enterprises in the world who have voluntarily shared their data to be part of this accessible and open digital platform. 

After the project launch in November 2024, the World Map project is expanding in 2025, the International Year of Cooperatives. This year we are growing our data to include new cooperatives in new regions, expanding the functionality of the map and seeking ways to shine a light on the invisible ecosystem of global cooperatives. 

This [Mykomap](https://digitalcommons.coop/mykomaps/) has been created by the [Digital Commons Co-operative](https://digitalcommons.coop/) using data published by [DotCooperation](https://identity.coop/) and many [international co-operative partners](https://worldmap.coop/our-data-partners/) across the world under the Open Data Commons Attribution Licence ([ODC-By v1.0](https://opendatacommons.org/licenses/by/1-0/).) By accessing the dataset you are agreeing to accept this license. 

The data was last updated in 31st March 2025. 

To see what what’s changed since the launch in Delhi last year, see the release notes [here](https://github.com/DigitalCommons/mykomap-monolith/releases/).

**Comment by ColmDC** [comment]
> This is now included in the 20250826 dataset.

To clariify, the correct text is being used, but it is not translated.

**Comment by codev** [comment]
Taking this one, updated description for @rogup [approach](https://github.com/DigitalCommons/mykomap-monolith/issues/85#issuecomment-2495517318) - now going to draft translations.

**Comment by codev** [comment]
@ColmDC - coming back to you for an opinion.

I got a bit confused by this but Marcel helped out.

The [canonical text](https://github.com/DigitalCommons/mykomap-monolith/issues/85#issuecomment-2912142773) has links in it so will require markdown rendering.

That will make the config.json quite difficult to edit.

So the choices are:

1. going back to the side by side files about.{lang}.md - easy to edit and can have links, headings etc, Powys does this I think, means a small back end change and four about files (one per language)
2. drop the links and simply the text so it fits in config.json with no links - might still be a bit more difficult to edit as ui vocab string than in an md files, config.json

**Comment by codev** [comment]
Also while I was looking I saw the latest data about.md doesn't have the latest test, only a couple of the datasets do - I need to investigate why some of them just get this autocreated looking about text "created on: 4/2/2026, 4:02:52 PM" - here are the datasets with the correct text:

```
DEV 2026-04-15 ~/data/dcc/cwm-test-data$ rg "Welcome to the Cooperative World Map"
datasets/cwm-20251212/about.md
1:Welcome to the Cooperative World Map!

datasets/cwm-20260314/about.md
1:Welcome to the Cooperative World Map!
```

**Comment by ColmDC** [comment]
We will need links in the About panel. DotCoop are updating the about text and it will include links to the worldmap.coop site and our website. They will share this new text with us this week. So it sounds like choice 1.

**Comment by codev** [comment]
Ok after discussion with @wu-lee I decided to keep everything in config.json but have added a script to convert from plain markdown files for ease of editing, I've translated the existing text using DeepL.

The files can be edited in the data-pipeline and will end up as markdown in the about page. Thanks also to Marcel for the Wednesday work session.

**Comment by codev** [comment]
I added the four markdown files, translated by DeepL here:

https://github.com/DigitalCommons/data-pipelines/pull/3/changes

The markdown files can be edited to produce a new config.json

DeepL seems to do quite well, I translated it back with Google Translate and it came out readable. The brand name 'Cooperative World Map' might need adding to a glossary if there is a set translation for each language.

**Comment by wu-lee** [comment]
Note, I'd suggest that the data-pipelines is probably not the right home for this `about.md` conversion.

Reason: although it does currently generate a MM dataset, it can only do that for a one branch of MM (or perhaps with alteration, a fixed number). Since each branch of MM can in principle use a different config.json format, and the specific list of PRs to generate for at any given moment in time is hard for data-pipelines to know, it doesn't seem the right place to do that.

Instead I'd propose that the dataset generation step is removed from `data-pipelines`. Instead, devise a data deployment process owned by each MM deployment, which can therefore know the `config.json` format it needs.  Perhaps this could be a script in the first instance.

For each dataset to deploy, this needs a set of URLs pointing at the upstream sources to use as inputs:
- a `config.json` (plus maybe one or more `about.md` files, although perhaps that detail should be dealt with upstream to reduce complexity here)
- an appropriate CSV file

These files would need to be downloaded, the conversion script run to create a MM dataset in the format required, and this deployed in the correct location for the map back-end. 

Errors in the process would need to be logged and fed back to the invoker.

The code for this would live in the `mykomap-monolith` project, probably along with the dataset conversion CLI in the back-end.

For this to work we need the CSV and `config.json` URLs stored somewhere they can be downloaded.

We already have our latest CSVs published: 
- the single-source cases are in dev.data.digitalcommons.coop, and
- the multi-source case is published as artifacts in the data-pipelines repository (i.e. the CWM; in future, maybe we might have more than one case?)

But what we don't have is a place where `config.json` files are published online. (Unless you count the `cwm-test-data` repo, which as its name suggests, is really just a stopgap means for deployment created for the CWM, and becoming large enough to be quite unwieldy to boot. It could be used as a source of JSON configs, but not very conveniently.) 

But that's outside the scope of this issue, so I won't talk about it any further.

**Comment by rogup** [comment]
@wu-lee Agreed on the points above. Isn't the MM generation step in data-pipelines just calling the script that lives in the monolith repo though? But maybe instead of this, it could be simpler to just call a webhook at the end of the data merging that triggers the monolith repo script to generate a new dataset, using a supplied CSV and config.json location

Also maybe generation of the config.json could also just live in this repo, since there could potentially be scenarios on the future where the data merging and config (e.g. vocab terms) are interdependent and parts of the config can be auto-generated by these scripts? Then it could just live as build artifacts in the same way as the CSVs

**Comment by ColmDC** [comment]
"Welcome to the Cooperative World Map! 

The Cooperative World Map (CWM) is an interactive and up-to-date digital map that aspires to unite and showcase all cooperative entities in the world. The CWM is a visual representation of all cooperative enterprises in the world who have voluntarily shared their data to be part of this accessible and open digital platform. 

After a year of expansion in 2025, the International Year of Cooperatives, we are aiming for even more in 2026. This year we explore more ways to grow the data to include new cooperatives in new regions, while enhancing the visual appearance and functionality of the map. 

This [Mykomap](https://digitalcommons.coop/mykomaps/) has been created by the [Digital Commons Co-operative](https://digitalcommons.coop/) using data published by [DotCooperation](https://identity.coop/) and many [international co-operative partners](https://worldmap.coop/our-data-partners/) across the world under the Open Data Commons Attribution Licence ([ODC-By v1.0](https://opendatacommons.org/licenses/by/1-0/).) By accessing the dataset you are agreeing to accept this license. 

The data was last updated on *** th April 2026."

**Comment by wu-lee** [comment]
> [@wu-lee](https://github.com/wu-lee) Agreed on the points above. Isn't the MM generation step in data-pipelines just calling the script that lives in the monolith repo though?

It is. Using (currently) only the dev branch of MM.  Thus in general the result won't be guaranteed to work on production, or any of the other PR branches we might want deploy data for.

> But maybe instead of this, it could be simpler to just call a webhook at the end of the data merging that triggers the monolith repo script to generate a new dataset, using a supplied CSV and config.json location

Maybe. Although that signal would need to go to all map deployments which need updating - there may not just be one. Because this is not something the data-pipeline project will easily be able to know about, by the same principle as above, I propose it would be better that the MM deployments manage their own (webhook) subscriptions to changes in their source data files, should they need that.

> Also maybe generation of the config.json could also just live in this repo, since there could potentially be scenarios on the future where the data merging and config (e.g. vocab terms) are interdependent and parts of the config can be auto-generated by these scripts? Then it could just live as build artifacts in the same way as the CSVs

Aren't config.json files necessarily hand-written for the most part?  Although I guess you could automate *parts* of that, things like choice of titles, options, the selections of which CSV fields to take and how to interpret them, these are very much a human choice, and not the result of an algorithm. But for now I would probably defer thinking about that and just take them as a given, to keep things simple.

**Comment by wu-lee** [comment]
Also: build artifacts have a limited lifespan, depending on their size and the GitHub plan, so we may have to find another way to publish artifacts more permanently if we want to deploy maps of older data.

**Comment by rogup** [comment]
> Maybe. Although that signal would need to go to all map deployments which need updating - there may not just be one. Because this is not something the data-pipeline project will easily be able to know about, by the same principle as above, I propose it would be better that the MM deployments manage their own (webhook) subscriptions to changes in their source data files, should they need that.

Are you referring to multiple Coolify deployments? Because we only have 2 MM backend deployments (dev-2 and prod-2), which are probably the main cases for this automatic updating. I feel like Coolify deployments for features wouldn't really have to stay updated with the latest data deployments except in edge cases where this is important, in which case the data rebuild could be triggered manually

> Aren't config.json files necessarily hand-written for the most part? Although I guess you could automate parts of that, things like choice of titles, options, the selections of which CSV fields to take and how to interpret them, these are very much a human choice, and not the result of an algorithm. But for now I would probably defer thinking about that and just take them as a given, to keep things simple.

Hmm yeah maybe a better place for the config.json files would actually be in the mykomap-monolith folder, along with the about.md files and the new conversion script. I feel like having another separate repo with these little bits just creates more faff and something that needs to be specifically pointed out when onboarding people - things that having a monolith is supposed to improve. I think there could be a 'branding' lib in the monolith that keeps map-specific stuff like this, along with the map-specific icon assets that we've recently been adding to the mykomap src code.

**Comment by wu-lee** [comment]
Just to recap an online video conversation with Marc just now...

There is still a problem in that the about.md is markdown, and although this can be embedded into JSON (as a term in vocab), trying to proof-read or edit it in this form will be highly inconvenient and error prone.

(See this [question](https://stackoverflow.com/questions/71321930/how-to-store-markdown-in-a-json-file) for example.)

So my current best suggestion for dealing with that without adding a stack of other problems is to just add YAML as a supported format. This can encode the same basic things as JSON, but in a more human-readable format.

So that instead of some JSON like this:

```
{
  "title": "Sample Document",
  "content": "# Header 1\nThis is some sample **Markdown** content.\n\n## Header 2\n- List item 1,  \"quote\"\n- List item 2\n\n[Link to Example](https://www.example.com)"
}
```

We could write YAML like this:

```
title: Sample Document
content: |
  # Header 1
  This is some sample **Markdown** content.

  ## Header 2
  - List item 1, "quote"
  - List item 2

  [Link to Example](https://www.example.com)
```

Given that, I think we can start using config.yaml files in place of config.json, and the about text can be a YAML inline block (as `content` is above).

---

*An aside...*

There is also potentially a problem managing our configs, in that a lot of our future maps will be sharing vocabularies (e.g. the ICA "Economic Activities" vocab is commonly used in several of the maps, as well as the CWM), and doing this naiively will create a lot of duplication in our configs. A big chunk will need to be cut and pasted into each, essentially manually, resulting in a maintenance headache if the vocab evolves: the same change will then need to be updated in many places.

What we probably need to do is to deduplicate that. Either by building the configs from upstream components (roughly as is done for Mykomap 3.x - when the data is normalised, one or more of SKOS vocabs published on vocabs.digitalcommons.coop can be read and used to generate a vocabs.json file the map then uses as part of its configuration) . And/or defining a way to reference external vocab definitions in our yaml files. (The latter might be a quicker fix? YAML doesn't support external includes natively but there are some ideas [here](https://www.w3tutorials.net/blog/yaml-add-link-reference-to-external-document/))

But this is a problem for the future, out of the scope of this issue.

**Comment by wu-lee** [comment]
To respond to Rohit... (Although this topic is out of the scope of this issue, and we should probably have a discussion in a call instead. This comment may deserve to be ignored by other readers.)

> Are you referring to multiple Coolify deployments? Because we only have 2 MM backend deployments (dev-2 and prod-2), which are probably the main cases for this automatic updating. I feel like Coolify deployments for features wouldn't really have to stay updated with the latest data deployments except in edge cases where this is important, in which case the data rebuild could be triggered manually

Those are probably the main cases, yes. Although I still think it's probably best to have the deployments notified rather than the data-pipeline. The deployments are going to need some smarts for this in any case, I think. But that's just my opinion.


> Hmm yeah maybe a better place for the config.json files would actually be in the mykomap-monolith folder, along with the about.md files and the new conversion script. I feel like having another separate repo with these little bits just creates more faff and something that needs to be specifically pointed out when onboarding people - things that having a monolith is supposed to improve. I think there could be a 'branding' lib in the monolith that keeps map-specific stuff like this, along with the map-specific icon assets that we've recently been adding to the mykomap src code.

Just to outline my thoughts and opinions on this...

The general rule of thumb I think this violates is that an application's config should not be baked into the code - these want to be separate concerns. 

So I don't think the `about.md` wants to live there either, by the same principle. And the normalisation of Powys data *really* doesn't belong in the mykomap application repo - it's coupled to the world of Powis project decisions, not mykomap development. Mykomap should be the dumb unopinionated recipient of data, a general principle to keep it simple to understand. Thus we should aim not be hardcoding Powis domain-knowledge into it, as that is creating technical debt. Again - an opinion, of a departing developer to boot - but happy to discuss and be convinced otherwise.

As an aside, and this is a different, if related, point again: I think we're also abusing git for deploying our datasets in `cwm-test-data`. Fine as a stopgap, but increasingly bad as it grows, it doesn't scale well. My thinking is influenced by this article, ["Package Managers keep using git as a database"](https://nesbitt.io/2025/12/24/package-managers-keep-using-git-as-a-database.html), which has some good points.

My suggestion would instead be to put the `config.json` files into a public web folder, as we do for `standard.csv` *et al.* And make sure all of that is backed up.  (I'd suggest add version numbers too, for configs and probably the data too. e.g. `https://dev.data.digitalcommons/powis/v1/config.json`, `https://dev.data.digitalcommons/powis/v2/config.json`, etc.)

**Comment by ColmDC** [comment]
About box text modified a little:
"Welcome to the Cooperative World Map! 

The Cooperative World Map (CWM) is an interactive and up-to-date digital map that aspires to unite and showcase all cooperative entities in the world, and is the forward facing part of the [Cooperative World Data Project.](https://worldmap.coop/) The CWM is a visual representation of all cooperative enterprises in the world who have voluntarily shared their data to be part of this accessible and open digital platform.  

After a year of expansion in 2025, the International Year of Cooperatives, we are aiming for even more in 2026. This year we explore more ways to grow the data to include new cooperatives in new regions, while enhancing the visual appearance and functionality of the map. 

This [Mykomap](https://digitalcommons.coop/mykomaps/) has been created by the [Digital Commons Co-operative](https://digitalcommons.coop/) using data published by [DotCooperation](https://identity.coop/) and many [international co-operative partners](https://worldmap.coop/our-data-partners/) across the world under the Open Data Commons Attribution Licence ([ODC-By v1.0](https://opendatacommons.org/licenses/by/1-0/).) By accessing the dataset you are agreeing to accept this license. 

The data was last updated on *** th April 2026. 

To see the website home page of this project, please go to [worldmap.coop](https://worldmap.coop/)

**Comment by codev** [comment]
Ok, I'm still getting up to speed but I think I understand most of this conversation now.

I think moving the config file to the data server and fetching it during the compile to be fetched is outside of the scope of this issue and should be scoped out separately.

I'm happy to switch to yaml if @wu-lee thinks that is better than trying to bake md files into json.

So I will make mykomaps look for config.yaml - fall back to config.json and alter the data pipeline to copy config.yaml and update the translations with the latest from @ColmDC https://github.com/DigitalCommons/mykomap-monolith/issues/85#issuecomment-4295223372 and add translations

Powys can stay as it is using the about.md fallback but needs to be brought in during the next phase of moving config files which I can look at in #271

**Comment by codev** [comment]
Ok having spent some time working on it yesterday I think this issue is a lot bigger if we're switching the configs to yaml - so I'll open a new issue to discuss it there.

I'm not sure what to move it to on the megaboard @lin-d-hop if it is blocked waiting on a new issue.
